fx-src/FirefoxStorage.js

```diff
@@ -197,6 +202,39 @@ export default class FirefoxAdapter extends BaseAdapter {
     });
   }
 
+  loadDump(records) {
+    let connection = this._connection;
+    let collection_name = this.collection;
+    return Task.spawn(function* () {
+      yield connection.executeTransaction(function* doImport() {
+        for (let record of records) {
+          const params = {
+            collection_name: collection_name,
+            record_id: record.id,
+            record: JSON.stringify(record)
+          };
+          yield connection.execute(statements.importData, params);
```

Comment:
At this point, all conflicting records have been deleted and the incoming records inserted.